grype/version/constraint_unit.go

```diff
@@ -3,53 +3,44 @@ package version
 import (
 	"fmt"
 	"regexp"
+	"strconv"
 	"strings"
+
+	"github.com/anchore/grype/internal"
 )
 
 // operator group only matches on range operators (GT, LT, GTE, LTE, E)
 // version group matches on everything except for whitespace and operators (range or boolean)
-var constraintPartPattern = regexp.MustCompile(`(?P<operator>[><=]*)\s*(?P<version>[^<>=\s,|]+)`)
+var constraintPartPattern = regexp.MustCompile(`\s*(?P<operator>[><=]*)\s*(?P<version>.+)`)
```

Comment:
not sure I follow why we allow any number of leading whitespaces

Reply:
I changed the expectations of what input can be given to constraintUnit, previously is was not possible to provide input with leading spaces, now it is (there is a matching test case to cover this as well).